tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py

```diff
@@ -191,36 +209,230 @@ def test_delete_registered_model_tag_unsupported(store):
+        )
+        store.create_model_version(name=model_name, source=source)
+        # Verify that s3 artifact repo mock was called with expected args
+        s3_artifact_repo_class_mock.assert_called_once_with(
+            artifact_uri=storage_location,
+            access_key_id=access_key_id,
+            secret_access_key=secret_access_key,
+            session_token=session_token,
+        )
+        mock_artifact_repo.log_artifacts.assert_called_once_with(local_dir=source, artifact_path="")
+        _assert_create_model_version_endpoints_called(
+            request_mock=request_mock, name=model_name, source=source, version=version
+        )
+
 
-    # test optional fields
-    run_id = uuid.uuid4().hex
-    description = "version description"
-    store.create_model_version(
-        "model_1",
-        "path/to/source",
-        run_id,
-        description=description,
+def test_create_model_version_azure(store, tmp_path):
+    storage_location = "abfss://filesystem@account.dfs.core.windows.net"
+    fake_sas_token = "fake_session_token"
+    temporary_creds = TemporaryCredentials(
+        azure_user_delegation_sas=AzureUserDelegationSAS(sas_token=fake_sas_token)
     )
-    _verify_requests(
-        mock_http,
-        "model-versions/create",
-        "POST",
-        CreateModelVersionRequest(
-            name="model_1",
-            source="path/to/source",
-            run_id=run_id,
-            description=description,
-        ),
+    with mock.patch("mlflow.utils.rest_utils.http_request") as request_mock, mock.patch(
+        "mlflow.store.artifact.azure_data_lake_artifact_repo.AzureDataLakeArtifactRepository"
+    ) as adls_artifact_repo_class_mock:
+        mock_adls_repo = mock.MagicMock(autospec=AzureDataLakeArtifactRepository)
+        adls_artifact_repo_class_mock.return_value = mock_adls_repo
+        source = str(tmp_path)
+        model_name = "model_1"
+        version = str(1)
+        request_mock.side_effect = get_request_mock(
+            name=model_name,
+            version=version,
+            temp_credentials=temporary_creds,
+            storage_location=storage_location,
+            source=source,
+        )
+        store.create_model_version(name=model_name, source=source)
+        adls_artifact_repo_class_mock.assert_called_once_with(
+            artifact_uri=storage_location, credential=ANY
+        )
+        adls_repo_args = adls_artifact_repo_class_mock.call_args_list[0]
+        credential = adls_repo_args[1]["credential"]
+        assert credential.signature == fake_sas_token
+        mock_adls_repo.log_artifacts.assert_called_once_with(local_dir=source, artifact_path="")
+        _assert_create_model_version_endpoints_called(
+            request_mock=request_mock, name=model_name, source=source, version=version
+        )
+
+
+@pytest.mark.parametrize(
+    "create_args",
+    [
+        ("name", "source"),
+        ("name", "source", "description", "run_id"),
+    ],
+)
```

Comment:
Good question, it's just because we only need to test common logic for supplying optional parameters to create a model version against a single cloud (the logic is the same across clouds). Open to parametrizing the other tests too, I just didn't want to complicate them

Reply:
Makes sense!